php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -746,11 +750,15 @@ class NDB_BVL_Instrument extends NDB_Page
             $values = Utility::nullifyEmpty($values, $key);
         }
         $db = \Database::singleton();
-        $db->update(
-            $this->table,
-            $values,
-            array('CommentID' => $this->getCommentID())
-        );
+        if ($this->jsonData !== true) {
+            // If the instrument is saving as JSON into the Data column, the
+            // table may not exist, so don't try and update it.
+            $result = $db->update(
+                $this->table,
+                $values,
+                array('CommentID' => $this->getCommentID())
+            );
+        }
 
         // Extract the old data and merge it with what was submitted so that we
         // don't overwrite data from other pages.
```

Comment:
this comment is for line 771

The current SQL column type is [TEXT](https://github.com/aces/Loris/blob/111bb660c9695cb4e5c6937323f677933ade0477/SQL/0000-00-00-schema.sql#L400)... its strange that a NULL text field is returning a "null" string. I would think it's an issue with saving where "null" is being submitted instead of NULL in the update

Reply:
It is strange, but it's the behaviour I'm was getting and I don't see what alternative there is to having that check.